### PR TITLE
feat: add no-invalid-named-grid-areas rule

### LINKED ISSUE #158: New Rule: no-invalid-named-grid-areas
### Rule details

Disallow invalid named grid areas.

### What type of rule is this?

Warns about a potential problem

### Example code

```css
a { grid-template-areas: "a a a"
                         "b b b b"; }
```

### Prior Art

[named-grid-areas-no-invalid](https://stylelint.io/user-guide/rules/named-grid-areas-no-invalid/) from stylelint

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### Additional comments

For a named grid area to be valid, all strings must define:

- the same number of cell tokens
- at least one cell token

And all named grid areas that spans multiple grid cells must form a single filled-in rectangle.

### PR DESCRIPTION
**Pull request by xbinaryx**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request?

This PR adds a new rule `no-invalid-named-grid-areas` to enforce valid named grid area definitions in CSS Grid templates. The rule helps prevent common mistakes when using grid-template-areas and related properties by checking that named grid areas are properly defined according to the CSS Grid specification.

#### What changes did you make? (Give an overview)

Added the  `no-invalid-named-grid-areas` rule, along with documentation and tests.

#### Related Issues

Fixes #158

### CODE REVIEW

**Review by nzakas — DISMISSED** [review]
LGTM. Thanks! Would like another review before merging.

File: src/rules/no-invalid-named-grid-areas.js [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+						}
+
+						const row = trimmedValue.split(" ").filter(Boolean);
+						grid.push(row);
+
+						if (firstRowLen === null) {
+							firstRowLen = row.length;
+						} else if (row.length !== firstRowLen) {
+							unevenNodes.push(stringNode);
+						}
```
Comment (snitin315) [inline src/rules/no-invalid-named-grid-areas.js:140]: Right now we don't report the `"a b"` line, but I think we should:

```css
/* eslint css/no-invalid-named-grid-areas: "error" */

.grid {
	grid-template-areas:
  "a b" // No error, but shouldn't we report this as well since this is also uneven and we don't know if the intention is to have 2,3, or 4 cells
  "a b c" //  Error: Grid area strings must have the same number of cell tokens 
  "a b c d"; //  Error: Grid area strings must have the same number of cell tokens 
}
```

Comment (snitin315) [reply src/rules/no-invalid-named-grid-areas.js:140]: Because the grid is inconsistent as a whole.

File: src/rules/no-invalid-named-grid-areas.js [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+					nonRectErrors.forEach(({ name, row }) => {
+						const stringNode = stringNodes[row];
+						context.report({
+							node: stringNode,
+							messageId: "nonRectangularGridArea",
+							data: {
+								name,
+							},
+						});
+					});
```
Comment (snitin315) [inline src/rules/no-invalid-named-grid-areas.js:173]: Here we report an error on line 5, but I think we should report line 6:

Screenshot 2025-06-21 at 9 48 24 PM

Comment (snitin315) [reply src/rules/no-invalid-named-grid-areas.js:173]: Here error in both rows but we report only one:

Screenshot 2025-06-21 at 9 50 25 PM

File: src/rules/no-invalid-named-grid-areas.js [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+					for (const stringNode of stringNodes) {
+						const trimmedValue = stringNode.value.trim();
+
+						if (trimmedValue === "") {
+							emptyNodes.push(stringNode);
+							continue;
+						}
+
+						const row = trimmedValue.split(" ").filter(Boolean);
```
Comment (snitin315) [inline src/rules/no-invalid-named-grid-areas.js:133]: Apparently, the escaping whitespace is valid syntax:

Screenshot 2025-06-21 at 9 58 33 PM

```css
grid-template-areas:
  "a \ a a" // Throws error but is valid
  "b c c"
  "b c c";
```

File: tests/rules/no-invalid-named-grid-areas.test.js [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+
+const ruleTester = new RuleTester({
+	plugins: {
+		css,
+	},
+	language: "css/css",
+});
+
+ruleTester.run("no-invalid-named-grid-areas", rule, {
+	valid: [
```
Comment (snitin315) [inline tests/rules/no-invalid-named-grid-areas.test.js:27]: Let's add some valid test cases with whitespace also:

```css
grid-template-areas:
  "  a b c "
  "a b c";
  
grid-template-areas:
  "header     header"
  "main       sidebar";
```

File: src/rules/no-invalid-named-grid-areas.js (hunk 1 above)
Comment (nzakas) [reply src/rules/no-invalid-named-grid-areas.js:140]: For most linting rules we only flag the first pattern that is unexpected. In this example, the second row is unexpected because of the first row setting the expectation that a row has two cells. So I think the current behavior is inline with how we generally make lint rules work.

File: src/rules/no-invalid-named-grid-areas.js (hunk 2 above)
Comment (nzakas) [reply src/rules/no-invalid-named-grid-areas.js:173]: As with my other comment, I think it's correct to only error on the second row here.

File: src/rules/no-invalid-named-grid-areas.js (hunk 3 above)
Comment (xbinaryx) [reply src/rules/no-invalid-named-grid-areas.js:133]: I couldn't reproduce this. Can you double-check?

File: src/rules/no-invalid-named-grid-areas.js (hunk 2 above)
Comment (snitin315) [reply src/rules/no-invalid-named-grid-areas.js:173]: Agreed, but here we are reporting the first row instead of 2nd in both cases.

File: src/rules/no-invalid-named-grid-areas.js (hunk 1 above)
Comment (snitin315) [reply src/rules/no-invalid-named-grid-areas.js:140]: Makes sense 👍🏻

File: src/rules/no-invalid-named-grid-areas.js (hunk 3 above)
Comment (snitin315) [reply src/rules/no-invalid-named-grid-areas.js:133]: Hmm, can't reproduce anymore.

File: src/rules/no-invalid-named-grid-areas.js (hunk 2 above)
Comment (xbinaryx) [reply src/rules/no-invalid-named-grid-areas.js:173]: @snitin315 Actually, it does report the second row in both cases. Could you please double-check?

Comment (snitin315) [reply src/rules/no-invalid-named-grid-areas.js:173]: @xbinaryx Can you please add a corresponding test cse?

File: tests/rules/no-invalid-named-grid-areas.test.js [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+				{
+					messageId: "nonRectangularGridArea",
+					data: { name: "a" },
+					line: 1,
+					column: 32,
+					endLine: 1,
+					endColumn: 37,
+				},
+			],
+		},
```
Comment (snitin315) [inline tests/rules/no-invalid-named-grid-areas.test.js]: Can we format it like https://github.com/eslint/css/pull/169#discussion_r2160075776 to clearly see which row is actually being reported

```css
grid-template-areas: 
     "a a"
      "a .";
```

It should report the 2nd row.

**Review by nzakas — APPROVED** [review]
LGTM. Would like @snitin315 to verify before merging.

**Review by snitin315 — APPROVED** [review]
LGTM, thanks!
